### DISCUSSION TITLE
Realtime action initiator

### DISCUSSION THREAD

**Discussion by synapse** [pocketbase/pocketbase#770] [Q&A] [answered]
Hey @ganigeorgiev is there a way of knowing who did the actual action? I would like to skip updating my local data in case the current user did the action. Let me explain my use case:

1. I'm listening to changes on a certain collection, using this approach [here](https://github.com/pocketbase/pocketbase/discussions/308#discussioncomment-3389399)
2. my client app is adding or updating collection items (this is also triggering the realtime update event on that collection)
3. because of the realtime event being trigger the client app reloads the collection (the last step is useless since it was the current user doing the action) - it would be nice to have the initiator of the event in the SSE data

**Comment by ganigeorgiev** [comment, marked as answer]
> is there a way of knowing who did the actual action?

No, at least not directly. You can attach a `createdBy/updatedBy` or similar `user`/`relation` type field and send the current user/profile id when creating or updating the record and compare it with your local user in the realtime subscribe callback.

---

> it would be nice to have the initiator of the event in the SSE data

It could be added, but technically guests also could be allowed to listen to realtime changes so I'm not sure how this will look like. Additionally, when PocketBase is used as framework you can also broadcast a realtime event programatically and in this case you may not have a "initiator".

---

This of course depends on your specific use case, but personally I'd handle this entirely client-side by checking if the received realtime event record differs from the one you already have and trigger the collection records reload only if there was an actual change. This way you'll handle the cases even when you have your app opened in different devices/tabs by the same user.

Another alternative approach will be also to not trigger full reload of your collection records list at all, but instead manipulate the initially fetched list based on the realtime action. For example, if `e.action === 'create'` then `push` the  new record to your list if doesn't exist, if it was `e.action === 'update'` do an upsert, etc.

**Reply by synapse** [reply]
Thanks @ganigeorgiev, in my use case I might go with the first version (having created/update by field) since I'm using Electron, and the realtime update event handling is happening in the main process (which does not have access to all data).

As for your second solution, not sure what guests have to do with it, since it would be just a matter of having something like an `id` of whom did the action, in case it's a guest user that it would be empty.

Again thanks for the idea!

**Reply by ganigeorgiev** [reply]
> As for your second solution, not sure what guests have to do with it, since it would be just a matter of having something like an id...

How will you handle the update in your case if the initiator field was empty?

In general, it's a little more complicated unfortunately. While we can add `null`/empty string as value, there is also the case where an admin could initialize the action and there is no guarantee that the admin id will not be the same as the user one. Additionally in the upcoming v0.8.0 we will have support for different auth models (aka. users, clients, etc.) so we will have to add also the `collectionId` to the payload. Once again, this could be done, but in my opinion it is better handled either by a user defined field or, when possible, do the checks entirely client-side to handle the same-user-multi-tabs scenario.
